### ISSUE TITLE
Request for Comment: Replace edX Calculus sequence with MIT OCW

### ISSUE THREAD

**Issue by dougc85** [ossu/computer-science#838]
**Background:**

I'm currently finishing up MIT's Calculus 1B course on edX and I've found it not to be particularly well set up to teach students calculus. It's unclear to me who the intended audience is for the course, since it doesn't seem at all to be geared toward students new to calculus; many of the video lectures are short and lack context for the discussion of each new concept.  The video lectures are akin to someone slowly reading out of a textbook, and little effort is made to anticipate issues students might have and to provide elaboration and encourage understanding. 

Oddly enough, another course from MIT (with some of the same content) does significantly more to pedagogically engage with students, is less discouraging for newcomers to calculus, and is, I think, better in terms of instruction.

**Advantages of OCW over edX:**

**1.** EdX cuts many of the recitation videos, which were among the most helpful bits of the OCW course. The two (student?) instructors take time to walk the viewer through each step of a problem, and they help to fill in any gaps you may have had after watching just the lectures. 

**2.** EdX cuts much of the lecture content of OCW and replaces it with their own shorter lectures. I found the OCW lectures more helpful to a student like me who had not studied calculus before. The edX lectures move quicker and explain less. Additionally, the OCW lectures are videos of actual classes, so they are occasionally punctuated by students asking for clarifications about things that I myself wished I could ask about. 

**3.** OCW has a massive problem bank with answers to all of the hundreds of problems offered there. OCW assigns selective problems for you to work on, but if you need more practice, the extra problems are there for you. No such extra practice exists in the edX class. 

**4.** OCW provides free exams with solutions; edX's exams are behind a paywall

**5.** OCW is available year-round, while edX is only available in cohorts.  You cannot, for example, start the course sequence at any time; you must wait for it to open.  Right now, for example, you cannot start Calculus 1A and there is no date set for when it will be available.

**Advantages of edX over OCW:**

**1.**  edX focuses on more challenging problems.  

Assuming a student is prepared for such problems, they will learn more by completing them.  To give an example of how difficult the problems are, though, in my Calculus 1A session, there were strings of comments about how difficult a particular unit was, and they were being posted by commenters who clearly already knew calculus (and much more) quite well, by their own admission. If even these commenters were finding trouble with a whole unit's worth of problems, I was left wondering who the course was designed for? It seems not to have been designed for someone learning the material for the first time.

**2.** edX focuses more heavily on applications of calculus to other fields of study.  Again, if prepared for the content, students will learn more about how calculus is used in the real world.  Unfortunately, I do not believe edX adequately prepares learners to engage with its physics/probability content.

**3.** edX has a comments section which is frequented by the instructors. This definitely **is**, without qualification, an advantage that edX has.  The instructors are very knowledgeable, kind, and helpful.  That said, the tutors and other students on the OSSU Discord also seem to have these same qualities and would perhaps provide a suitable substitute for this feature.

**Comments made by other students on edX**

Here is a selection of comments from the end of unit 3 of the current session of Calculus 1B; many students here were leaving comments similar to my sentiments above.  Admittedly, some other students also commented on how much they loved the course, but to my eyes these comments were limited to learners who were returning to math after some absence and trying to brush up their skills, not posted so much by students actually learning the material.

"Original MITOCW lectures have saved me. Edx materials are well constructed, but they are like patchwork, and I lost coherency. Edx way must be good for learners having knowledge, they can get information efficiently, but for me, it did not work. Then I found original lectures, and wow! Original lectures are theoretical and consistent, telling me where I am. Important thing is, original lectures have contexts! Contexts are very important for me to construct things. Only essence like edx is not enough."

"the gap between the exercises (which are already harder than what you would see in a typical calculus course) and the problem sets is so large as to be demoralizing even for a student with good math skills and a strong work ethic like myself. Maybe if I were at university taking physics and other technical courses in conjunction with this it would be a different story. I don't know."

"I'm doing fine at the regular class exercises, but when it comes to the Problem Sets, I've been consistently whacked. I think I read somewhere that people usually study from 5 to 10 hours a week for MITx Calculus programs, but for this last Problem Set I've been doing at least 3 hours a day. I almost gave up the course while doing this Problem Set 3A!"

"Try watching the lectures only in the OpenCourseware series, then memorize the basic concepts so they can be recalled immediately. At the end of the day I listen to these lectures only, they are more concise and accessible in OpenCourseware."

"I agree with watching the OCW videos that are very sequential."

**Proposal:**

For OSSU's calculus requirement, replace the edX Calculus sequence with MIT's OCW
https://ocw.mit.edu/courses/mathematics/18-01sc-single-variable-calculus-fall-2010/

**Comment by bradleygrant** [comment]
No objections.

**Comment by Alaharon123** [comment]
Something to note is that the edX/OLL version of this course has an optional section on limits, unlike the OCW version. Additionally, limits are not covered in the Khan Academy high school math sequence that OSSU currently recommends nor is it covered in ASU's precalculus course that is sometimes recommended on the Discord. So if the switch is made, which does sound like a good idea for all the reasons mentioned, there'd probably need to be a resource on limits added such as https://www.khanacademy.org/math/ap-calculus-ab/ab-limits-new/

**Comment by spamegg1** [comment]
I am taking Calculus 1A (to help out a student) on edX and I found it to be quite excellent.
Can't say anything about 1B or 1C yet.
The shorter videos are very good for online interactive pedagogy.
A lecture or recitation-length example is taken up and broken into 10+ interactive pieces of quizzes.
Each quiz holds you by the hand and carefully walks you through the understanding in very small steps.
A LOT OF EFFORT is made to anticipate what difficulties a student might face.
Often the student is led to intuitively figure out a difficulty (with non-penalizing multiple choice questions),
which is then immediately explained by a short video.
The discussion sections at the bottom of each page are quite active with lots of good questions and answers.

Granted, this style of learning and pedagogy is sure to frustrate some people.
I can see that in the discussion sections. Similar sentiments.
Although a lot of learners seem to be loving it.
I am guessing @dougc85 is one of those people whose natural learning style does not mesh very well with this.
There are students who cannot get through the longer lecture videos without falling asleep.
Personally I taught HUNDREDS of such students! 
Some would never show up to lectures at all, only to recitations.

So... I found Calculus 1A to be quite the opposite of @dougc85 's experience. 
To me it seems like they took a good quality lecture+recitation combo,
and figured out how to translate it to online interactive learning really well.

Personally I don't care either way... 
in fact I would prefer OCW because I don't want to deal with stupid "honor codes".
But I have to ask @dougc85 : 
why the replacement?
why does it have to be either/or?
Why either OCW or edX?

Can't we offer both, and say "take either according to your learning style/preference"?
And also "take OCW if edX is currently not active".
And also "here is OCW problem bank you might like."
I certainly do not want to rob students who learn better with shorter videos and online interactive style.

I would also like to ask @dougc85 : did you take Calculus 1A?
You say "... for someone like me who didn't learn calculus before"
Did you jump straight to Calculus 1B?
It's quite possible that this is unintended, and MIT prepared the courses to have a certain difficulty curve,
and 1A prepares you to get ready for the higher difficulty later.
I mean, it IS MIT after all.

**Comment by dougc85** [comment]
@spamegg1 

To a certain extent, I think we're just going to disagree about this, but let me try to address some of your points.

To start, yes I did take Calculus 1A, on differentiation. I think it would be a little ridiculous of me to suggest a replacement of a math curriculum I didn't take. 

As I've studied, I've watched all the OCW lectures and recitations and completed all the assigned problem sets.  I've also done all of the EdX material, and I've taken all the tests on Khan Academy as extra practice.  I actually think Sal, on Khan Academy, is the best teacher of all the above material, in terms of thoroughly explaining concepts, but the practice problems on KA are much, much easier than the MIT material, so I can see why the MIT courses would be preferred by OSSU.

I think the EdX 1A course starts very much as you describe, actually.  Not all of the content is as I described above.  The first two units were generally fairly thorough, and I didn't have much problem with them.

As the course progresses, though, I think this becomes less and less the case.  I disagree, in particular, with your assertion that "A LOT OF EFFORT is made to anticipate what difficulties a student might face.".  On the contrary, I think little to no effort has been made in this regard. The more difficult problems in the HW (starting, I think in unit 3...maybe 4) are chained together such that if you can't figure out the first part, you really can't do a whole sequence of problems. Further, the worked solutions were not enough for me to be able to understand what even was being done to solve the problem. 

This issue only magnifies as the course progresses and then is amplified further in 1B.  There is significantly less video content in the Integration section of the course (1B), and what there is, as I stated above, is about as dry as a read textbook and, again, made with seemingly little effort to anticipate problems students might face.

How far are you in the material, @spamegg1 ? And have you done all of the problems?  I wonder if you've reached the later sections I'm talking about.  

I suppose it's also possible that the course has been modified since I took it; I know they regularly ask for student feedback to improve their courses.  It could be the case that we're talking past each other, having each looked at a different version of the course.

As to some of your points:

-I actually agree with your suggestion that both courses be made known to learners.  I have many criticisms of the EdX course, but I am glad I went through it alongside OCW, as I think I have a firmer grasp of the material for it.

-EdX includes about 1/3 or fewer of OCW's recitation videos; definitely not all

-I don't agree with your framing of my argument as 'long lectures fit my learning style better than short content'.  As I said above, I actually think the Khan Academy videos do the best job of actually TEACHING (that is to say anticipating students' issues, explaining why we do this or that step of a problem, explaining the bigger picture of how things fit together) and those are short videos. 

-I maintain that students will be more likely to learn and finish calculus if they primarily follow OCW, in preference to EdX

**Comment by spamegg1** [comment]
@dougc85 Ah OK then.

I am close to finishing Calculus 1A and I do all of the problems.

Sorry if it came off as reframing your argument. I was simply guessing.
Sometimes I act like I can read people's mind :D I gotta stop doing that.
Because I taught so many students, I came across these learning-style related issues a lot.

I still maintain that they go to great lengths to take care of potential issues students might face.
You are correct that we are just going to disagree. 
I still believe this is somehow related to learning styles... but never mind. I'm just wrong.

It's a bit strange that you took Calculus 1A and you were mostly OK with it,
but the quality really started to drop in Calculus 1B,
and you suggest we replace *the whole sequence*.
Would it not be more natural to suggest replacing just Calculus 1B?
I wish you mentioned your Calculus 1A experience in your original post.

Can there be such a big difference in quality between 1A and 1B? It's hard to believe.
Do you think it's going to get even worse in 1C? 

I can understand the frustration of not getting the first part of a chain of exercises.
However for a lot of people who *do* get the first part, this works out really great.
So it's a bit of a pedagogical risk they are taking.
You'll see this a lot in other MIT courses too, like MIT6.00.1x Intro CS course we have.
When it works, it works great; when it doesn't work it fails catastrophically.

We get a lot of issues opened by learners who want to replace one thing for another.
I'm guessing that if your suggestion goes through, a few months later someone else will open an issue
"bring back edX Calculus".
They will probably say something like "OCW long lectures are really boring I can't get through them"

Anyway, this is a request for comment, and I commented. 
I still hope you adjust your suggestion to include edX versions though.

**Comment by bradleygrant** [comment]
My recommendation is that we include both course sources as top-level options (i.e. not one main suggestion and one alternate suggestion).

**Comment by dougc85** [comment]
@spamegg1 
You raise a good question regarding why we should replace the whole sequence. I have two different responses to that.

-The simplest answer is just that the courses are structured differently.  EdX is three separate calculus courses (1A, 1B, 1C) where as OCW is just one long calculus course covering all of the same material.  I think it would be a bit strange and overly complicated to suggest jumping from the end of 1A to week 5.5 (or whatever) of OCW.

-The other answer is that, while I think 1B is less helpful to inexperienced students than 1A, I don't think either does as good a job as OCW, so I would still recommend replacing the whole thing.  

I'm not sure I would feel as strongly critical of EdX if I hadn't been doing the two courses simultaneously, but since I did, I found myself often comparing the two courses in my mind. 

In the end, my recommendation remains the same.  I think there would be more clarity to just having a single recommended course, which I think should be OCW.  I don't know if there's space on the front page for something like "go here for extra practice". If there is, I'd support adding EdX to a section such as that, and Khan Academy, for that matter, which I think also provides valuable instruction.

Whatever the OSSU community decides to do, I'll respect the decision.  There's good content to be found in all of these courses, and some of y'all (including you, @spamegg1) obviously have a lot more mathematics experience than me.  That said, my experience as a learner has been that OCW has been head and shoulders above EdX in terms of clarity and thoroughness of explanations.

I don't know what more I can add, since I think I'm just repeating myself at this point. Thanks for the good discussion.

**Comment by spamegg1** [comment]
OK, just finished MIT Calculus 2 (Integration) on edX.
Once again great course. Very well fitted for online learning, easy small steps.
I found it to be even clearer and easier to understand than Calculus 1.
(Calculus 1, especially the Approximations chapter, was a bit cryptic and hard to follow. It was also too long.
They are basically trying to teach you power series, without teaching you power series...
That's just MIT being MIT I guess.)
The steps and videos were much easier to follow and connect.
There were also a lot of recitation style videos, with even more frequency than Calculus 1.

As for the "more focus on applications":
It's natural for Calculus, especially integration, to focus a lot on applications. That's what Calculus is for! Calculus has always been for physicists and engineers to use in real world applications, not for computer scientists. It's a good thing for a Calculus course to spend MORE time on applications, not less.

However I'm somewhat on your side. In the past I argued for removing Calculus from the curriculum. (I don't care either way anymore.)

**Comment by waciumawanjohi** [comment]
This RFC has not garnered comments in many months, it is time to resolve it!

Findings:
- There has been a healthy discussion about the merits of the approaches of the edx course and the OCW course. In the interim, two additional issues have been opened that point out that edX restricts access to the calculus course after the audit period. Both of these issues recommended using the MIT Open Learning Library (OLL) version. OLL is a relatively new option from MIT.
- Participants in this discussion have generally agreed with the idea of making students aware of the alternative MIT course.
- There has not been consensus on whether the single OCW course or the 3 individual courses are a better pedagogical fit.

Conclusion:
Availability trumps most any other concern; an excellent product that students cannot access is a poor match for OSSU. As a result, OSSU will no longer recommend the edX version of the course. The two remaining options (OLL and OCW) will both be highlighted to students. Given the subsequent issues recommending use of OLL, that will be listed as the primary, while OCW will be listed as the alternative.
